community/codegen/src/main/java/org/neo4j/codegen/MethodDeclaration.java

```diff
@@ -172,10 +211,63 @@ public TypeReference declaringClass()
         return parameters;
     }
 
+    public MethodDeclaration erased()
+    {
+        Map<String,TypeReference> table = new HashMap<>();
+        for ( TypeParameter parameter : typeParameters )
+        {
+            table.put( parameter.name(), parameter.extendsBound() );
+        }
+
+        TypeReference newReturnType = erase( returnType(), table );
+        Parameter[] newParameters = new Parameter[this.parameters.length];
+        for ( int i = 0; i < parameters.length; i++ )
+        {
+            Parameter parameter = parameters[i];
+            TypeReference erasedType = erase( parameter.type(), table);
+            newParameters[i] = param( erasedType, parameter.name() );
+        }
+        TypeReference[] newExceptions = new TypeReference[exceptions.length];
+        for ( int i = 0; i < exceptions.length; i++ )
+        {
+            newExceptions[i] = erase( exceptions[i], table );
+        }
+        String newName = name();
+        boolean newIsConstrucor = isConstructor();
+
+        return new MethodDeclaration( owner, newParameters, newExceptions, modifiers, typeParameters )
```

Comment:
Since this is declared in a method, it has a reference to the object that it was instantiated from (whether it uses it or not). I don't know if it is harmful (haven't done the full analysis yet), but it might keep that object alive for longer than needed.